cmd/build/v2/services.go

```diff
@@ -94,8 +94,17 @@ func (bc *OktetoBuilder) checkServiceToBuild(service string, manifest *model.Man
 	if isStack && okteto.IsOkteto() && !bc.Registry.IsOktetoRegistry(buildInfo.Image) {
 		buildInfo.Image = ""
 	}
-	buildHash := bc.hasher.hashService(buildInfo)
-	imageChecker := getImageChecker(buildInfo, bc.Config, bc.Registry, bc.ioCtrl.Logger())
+
+	var err error
+	var buildHash string
+	if bc.smartBuildCtrl.IsEnabled() {
+		buildHash, err = bc.smartBuildCtrl.GetBuildHash(buildInfo)
+		if err != nil {
+			bc.ioCtrl.Logger().Infof("error getting build hash: %s", err)
+		}
+	}
+
```

Comment:
this block is repeated. Should we encapsulate it in a func?

Reply:
I think it's ok to have it explicit in order to now that is calling the smartbuild. I can move it into a new function but I think it's more readable like this